test/server_search.cpp

```diff
@@ -736,11 +806,70 @@ struct TestData
     );
   }
 
+    std::string expectedXmlHeader() const
+    {
+      std::string header = R"(<title>Search: PATTERN</title>
+    <link>URL</link>
+    <description>Search result for PATTERN</description>
+    <opensearch:totalResults>RESULTCOUNT</opensearch:totalResults>
+    <opensearch:startIndex>FIRSTRESULT</opensearch:startIndex>
+    <opensearch:itemsPerPage>ITEMCOUNT</opensearch:itemsPerPage>
+    <atom:link rel="search" type="application/opensearchdescription+xml" href="/ROOT/search/searchdescription.xml"/>
+    <opensearch:Query role="request"
+      searchTerms="PATTERN"LANGQUERY
+      startIndex="FIRSTRESULT"
+      count="ITEMCOUNT"
+    />)";
+
+      const auto realResultsPerPage = resultsPerPage?resultsPerPage:25;
+      const auto url = makeUrl(query + "&format=xml", firstResultIndex, realResultsPerPage);
+      header = replace(header, "URL", replace(url, "&", "&amp;"));
+      header = replace(header, "FIRSTRESULT", to_string(firstResultIndex));
+      header = replace(header, "ITEMCOUNT",  to_string(realResultsPerPage));
+      header = replace(header, "RESULTCOUNT", to_string(totalResultCount));
+      header = replace(header, "PATTERN",     getPattern());
+      auto queryLang = getLang();
+      if (queryLang.empty()) {
+        header = replace(header, "LANGQUERY", "");
+      } else {
+        header = replace(header, "LANGQUERY", "\n      language=\""+queryLang+"\"");
+      }
+      return header;
+    }
+
+    std::string expectedXmlResultsString() const
+    {
+      if ( results.empty() ) {
+        return "\n    ";
+      }
+
+      std::string s;
+      for ( const auto& r : results ) {
+        s += "\n    <item>\n";
+        s += maskSnippetsInXmlSearchResults(r.getXml());
```

Comment:
I wonder if we could generate xml (and html) without snippets instead of generated a full xml and mask the snippet.

Reply:
That would mean having a special mode for the /search endpoint (e.g. `/search?snippets=no`), however I don't think that adding it for testing alone makes sense. Besides, we ignore snippets only at this point. Later snippets are validated too.

Reply:
I was speaking about the generation of the expected result. We would still have to mask the snippet in the actual results of the tested server.

Reply:
That makes sense. But not to an extent to pursue it if this is going to be the only thing to fix about this PR.

Reply:
Frankly I would rather change `TestData::expectedXmlResultsString()` to return the real expected response body and perform the masking out of snippets at the check site. That will eliminate some confusion implanted in the current code.

Reply:
> But not to an extent to pursue it if this is going to be the only thing to fix about this PR.

Agree with that. 

> Frankly I would rather change TestData::expectedXmlResultsString() to return the real expected response body and perform the masking out of snippets at the check site.

This is the other way around. Feel free to do it (but not in this PR, I approve and merge it)